scripts/Muon/GUI/MuonAnalysis/loadrun/load_run_model.py

```diff
@@ -0,0 +1,118 @@
+from __future__ import (absolute_import, division, print_function)
+
+import os
+
+import mantid.simpleapi as mantid
+from mantid.api import WorkspaceGroup
+
+from Muon.GUI.Common.muon_load_data import MuonLoadData
+
+
+def is_workspace_group(workspace):
+    return isinstance(workspace, WorkspaceGroup)
+
+
+def get_run_from_multi_period_data(workspace_list):
+    """Checks if multi-period data has a single consistent run number and returns it, otherwise raises ValueError."""
+    runs = [ws.getRunNumber() for ws in workspace_list]
+    unique_runs = list(set(runs))
+    if len(unique_runs) != 1:
+        raise ValueError("Multi-period data contains >1 unique run number.")
+    else:
+        return unique_runs[0]
+
+
+def exception_message_for_failed_files(failed_file_list):
+    return "Could not load the following files : \n - " + "\n - ".join(failed_file_list)
+
+
+class LoadRunWidgetModel(object):
+    """Stores info on all currently loaded workspaces"""
+
+    def __init__(self, loaded_data_store=MuonLoadData()):
+        # Used with load thread
+        self._filenames = []
+
+        self._loaded_data_store = loaded_data_store
+        self._current_run = None
+
+    # TODO : remove the need for this function
```

Comment:
when can we do this?

Reply:
I think I may just remove the `TODO`. The reason it was there was the function being a bit "hacky". But it is really the only choice if I want to be able to remove the "last-but-one" data that was added to the `MuonLoadData` class.